crates/starknet-devnet/tests/test_abort_blocks.rs

```diff
@@ -0,0 +1,239 @@
+pub mod common;
+
+mod abort_blocks_tests {
+    use hyper::Body;
+    use serde_json::{json, Value};
+    use server::api::json_rpc::error::ApiError;
+    use starknet_rs_core::types::{FieldElement, TransactionStatus};
+    use starknet_rs_providers::Provider;
+    use starknet_types::rpc::transaction_receipt::FeeUnit;
+
+    use crate::common::background_devnet::BackgroundDevnet;
+    use crate::common::utils::get_json_body;
+
+    static DUMMY_ADDRESS: u128 = 1;
+    static DUMMY_AMOUNT: u128 = 1;
+
+    async fn abort_blocks(devnet: &BackgroundDevnet, starting_block_hash: &Value) -> Vec<Value> {
+        let abort_blocks = devnet
+            .post_json(
+                "/abort_blocks".into(),
+                Body::from(json!({ "starting_block_hash": starting_block_hash }).to_string()),
+            )
+            .await
+            .unwrap();
+        let aborted_blocks = get_json_body(abort_blocks).await;
+
+        aborted_blocks["aborted"].as_array().unwrap().clone()
+    }
+
+    async fn assert_block_rejected(devnet: &BackgroundDevnet, block_hash: &Value) {
+        let block_after_abort = &devnet
+            .send_custom_rpc(
+                "starknet_getBlockWithTxHashes",
+                json!({
+                    "block_id": {"block_hash": block_hash},
+                }),
+            )
+            .await["result"];
+        assert_eq!(block_after_abort["status"], "REJECTED".to_string());
+    }
+
+    #[tokio::test]
+    async fn abort_latest_block() {
+        let devnet: BackgroundDevnet =
+            BackgroundDevnet::spawn_with_additional_args(&["--state-archive-capacity", "full"])
+                .await
+                .expect("Could not start Devnet");
+
+        devnet.create_block().await.unwrap();
+        let first_block = &devnet
+            .send_custom_rpc("starknet_getBlockWithTxHashes", json!({ "block_id": "latest" }))
+            .await["result"];
+
+        devnet.create_block().await.unwrap();
+        let second_block = &devnet
+            .send_custom_rpc("starknet_getBlockWithTxHashes", json!({ "block_id": "latest" }))
+            .await["result"];
+
+        let aborted_blocks = abort_blocks(&devnet, &second_block["block_hash"]).await;
+        assert_eq!(aborted_blocks[0], second_block["block_hash"]);
```

Comment:
Here and in few other places in this file, I think it is better to avoid indexing. Because - what if there are more elements in the tested array/vector? It's better to assert entire object equality:
```rust
assert_eq!(aborted_blocks, vec![latest_block["block_hash"].clone()]);
```

Reply:
Fixed in a2eeb56b6038ee38acf4ef02f54ee1aa2db11a38